File: Runtime/StateClassCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' The state class collection class
''' </summary>
''' <remarks></remarks>
Friend Class StateClassCollection
    Inherits KeyedCollection(Of Integer, StateClass)

    Protected Overrides Function GetKeyForItem(ByVal item As StateClass) As Integer
        Return item.Id
    End Function

End Class

File: src/Runtime/StateClassCollection.cs
﻿// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2021 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class StateClassCollection : KeyedCollection<int, StateClass>
    {
        protected override int GetKeyForItem(StateClass item)
        {
            return item.Id;
        }
    }
}
